### PR TITLE
jQuery should be a peerDependency

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
       "url": "http://opensource.org/licenses/Apache-2.0"
     }
   ],
-  "dependencies": {
+  "peerDependencies": {
```
Comment (itsjavi) [inline package.json]: please note that this behaves different in NPM older than v3. We should also keep it under `dependencies`. I added a commit to fix that.

Comment (graingert) [inline package.json]: No it should only be a peerDependency

**Comment by itsjavi** [comment]
@graingert I am trying to fix the PR, can you stop closing this issue? thanks ;)

**Comment by graingert** [comment]
@itsjavi jquery should only be required as a peerDependency. If you require it as both it breaks the whole point of peer dependencies

**Comment by itsjavi** [comment]
@graingert if we require it only as `peerDependency` we may break user's worfkflow having npm < 3.

**Comment by graingert** [comment]
you should release this on a new major version

**Comment by itsjavi** [comment]
@graingert I wouldn't jump to v3 only because of NPM compatibility. This kind of backwards incompatible changes are better in the develop branch (or any other WIP branches in general), not in master.

**Comment by graingert** [comment]
Sorry I was assuming github flow

**Comment by graingert** [comment]
peerDependency works fine on its own in npm v2 and v3 but will break when
duplicate packages exist

On 28 Dec 2016 15:50, "Thomas Grainger" <tagrain@gmail.com> wrote:

Sorry I was assuming github flow

**Comment by itsjavi** [comment]
@graingert thanks, It will be done in v3

**Pull request closed without merging**